andela/apollo-ah-backend
# #165008124 add image support for articles

#### What does this PR do?

- Adds image upload support to articles table

#### Description of Task to be completed?

- add image column to article schema and migration file
- add better slug constructor for update articles

#### How should this be manually tested?

- follow set up instruction from this [link](https://github.com/andela/apollo-ah-backend/tree/chore/164859311/update-readme#set-up)

- create a user from the ```/users/``` route

- make sure payload is in this format
```
{
  "email": "dbrandy@email.com",
  "password": "!23ijdkdhfhf",
  "username": "username101",
}
```
- visit the create articles route ```/articles/``` to create an article

- make sure you provide the payload in this format
```
{
  "title": "bvdd",
  "body": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9",
  "description": "hddkjkd",
  "categoryId": 1,
  "image": "link.to.image.url",
  "tagList": ["tech"]
}
```
- the image url will be provided in the response

#### Any background context you want to provide?

- Image upload will be handled from the frontend. The backend only accepts the resulting url.

#### What are the relevant pivotal tracker stories?

[#165008124](https://www.pivotaltracker.com/story/show/165008124)
